ocaml/idl/datamodel_pool.ml

```diff
@@ -1490,6 +1490,11 @@ let t =
         ; field ~qualifier:DynamicRO ~lifecycle:[] ~ty:Bool
             ~default_value:(Some (VBool false)) "update_sync_enabled"
             "Whether periodic update synchronization is enabled or not"
+        ; field ~qualifier:DynamicRO ~lifecycle:[]
+            ~ty:(Map (String, String))
+            ~default_value:(Some (VMap [])) "recommendations"
+            "An (string, string) map of recommended values and ranges for \
```

Comment:
I don't think the document string gives enough relevant information on what the field's purpose is and how it should be used. For example, what recommends these properties? xapi, xenserver, the client?

It also shows type information, which is shown elsewhere.

```suggestion
            "The recommended pool properties for clients to respect for optimal performance. e.g. max_vms : 15"
```

Reply:
I think we should spell out that these values are never used by xapi but created and used by an external client. For that reason I find it questionable to add these fields. We could add a general key/value store for a client to use that is under the control of the client. I also see potential for abuse by storing arbitrary large data (but that exists in other places as well).

Reply:
> I don't think the document string gives enough relevant information on what the field's purpose is and how it should be used. For example, what recommends these properties? xapi, xenserver, the client?

I think this is a general problem across many of the xapi fields. The doc is often just a repetition of the name of the field, which is kind of ponitless. We should have some guidelines on this when adding new fields, perhaps in that tutorial section?